packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/TaskNavButtonsContainer.spec.js

```diff
@@ -85,142 +115,117 @@ describe('TaskNavButtonsContainer', function () {
-
-      wrapper = shallow(
-        <TaskNavButtonsContainer.wrappedComponent
-          removeAnnotation={removeAnnotationSpy}
-          selectStep={selectStepSpy}
-          step={activeStep}
-          tasks={activeStepTasks}
-        />
-      )
-    })
-
-    afterEach(function () {
-      selectStepSpy.resetHistory()
-      removeAnnotationSpy.resetHistory()
+      classifierStore = setupMocks()
+      const store = { classifierStore }
+      wrapper = shallow(<TaskNavButtonsContainer store={store} />)
     })
 
-    it('should not call props.selectStep if there is not a previous step', function () {
+    it('should not go back if there is not a previous step', function () {
+      let activeStep = classifierStore.workflowSteps.active
+      expect(activeStep.stepKey).to.equal('S0')
       wrapper.instance().goToPreviousStep()
-      expect(selectStepSpy).to.have.not.been.called()
-    })
+      activeStep = classifierStore.workflowSteps.active
+      expect(activeStep.stepKey).to.equal('S0')
+    })
+
+    describe('when there is a previous step', function () {
+      before(function() {
+        // set an answer to the branching task question, so that step.next is set.
+        const classification = classifierStore.classifications.active
+        const singleChoiceAnnotation = classification.annotation({ taskKey: 'T0'})
+        singleChoiceAnnotation.update(0)
+        // push the first task to the history stack
+        wrapper.instance().goToNextStep()
+        wrapper.instance().goToPreviousStep()
+      })
 
-    it('should call props.selectStep when there is a previous step', function () {
-      const nextStep = Step.create({ stepKey: 'S1', taskKeys: ['T1'], previous: 'S0' })
-      wrapper.setProps({ getPreviousStepKey: getPreviousStepKeyStub, step: nextStep })
-      wrapper.instance().goToPreviousStep()
-      expect(selectStepSpy).to.have.been.called()
-      expect(selectStepSpy.withArgs('S0')).to.have.been.calledOnce()
-    })
+      it('should undo the most recent step', function () {
+        const { latest, steps} = classifierStore.annotatedSteps
+        expect(steps.size).to.equal(1)
+        expect(latest.step.stepKey).to.equal('S0')
+      })
 
-    it('should call props.removeAnnotation when there is a previous step', function () {
-      const nextStep = Step.create({ stepKey: 'S1', taskKeys: ['T1'], previous: 'S0' })
-      wrapper.setProps({ getPreviousStepKey: getPreviousStepKeyStub, step: nextStep })
-      wrapper.instance().goToPreviousStep()
-      expect(removeAnnotationSpy).to.have.been.called()
-      expect(removeAnnotationSpy.withArgs('T1')).to.have.been.calledOnce()
+      it('should select the previous step', function () {
+        const activeStep = classifierStore.workflowSteps.active
+        expect(activeStep.stepKey).to.equal('S0')
+      })
+
+      it('should remove annotations for the current task', function () {
+        const { annotations, steps } = classifierStore.annotatedSteps
```

Comment:
TODO: figure out how to remove orphaned annotations from `classification.annotations` when `clearRedo()` runs.